jib-maven-plugin fails if maven-war-plugin "warName" property is set

**Environment**:

- *Jib version:* 1.8.0, 1.7.0
- *Build tool:*  Maven 3.6.3
- *OS:* Fedora / Windows

**Description of the issue**:
Between 1.6.1 and 1.7.0, the behaviour of the jib-maven-plugin changed and no longer works if the WAR name is not according to the default (artifactID-version.war), but changed using the "warName" property of the maven-war-plugin.

**Expected behavior**:
Upgrading the jib-plugin from 1.6.1 to 1.7.0 does not change the result of the build.

**Steps to reproduce**:
I created a small repository that you can clone: https://github.com/padyx/jib-test-repository

Working configuration (initial repository state):

1. Alter the pom.xml File in _helloworld.ui.html.app.war_ and change the target image to a location you can push to. 
2. Execute the build as follows: `mvn -Djib.to.auth.username=xxxxx -Djib.to.auth.password=xxxxx -DskipTests -P docker clean package`  (Success)

Error if upgrading:
1. Change the jib-maven-plugin Version to 1.7.0 (or 1.8.0) in the helloworld/pom.xml File.
2. Rerun the build with the command above.


**`jib-maven-plugin` Configuration**: <!-- Delete this section if not used -->
```xml
<!-- From parent module (build / pluginManagement) -->
<plugin>
  <groupId>com.google.cloud.tools</groupId>
    <artifactId>jib-maven-plugin</artifactId>
    <version>1.7.0</version>
    <executions>
      <execution>
      <phase>package</phase>
      <goals>
        <goal>build</goal>
      </goals>
    </execution>
  </executions>
</plugin>

<!-- From build in actual module-->
<plugin>
  <groupId>com.google.cloud.tools</groupId>
  <artifactId>jib-maven-plugin</artifactId>
  <configuration>
    <from>
      <image>tomcat:9-jdk11</image>
    </from>
    <to>
      <image>padyx/testimage</image>
    </to>
  </configuration>
</plugin>
```

**Log output**: 
`[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.7.0:build (default) on project helloworld.ui.html.app.war: Obtaining project build output files failed; make sure you have compiled your project before trying to build the image. (Did you accidentally run "mvn clean jib:build" instead of "mvn clean compile jib:build"?): /home/chakotay/eclipse-workspace/helloworld/helloworld.ui.html.app.war/target/org.eclipse.scout.apps.helloworld.ui.html.war -> [Help 1]`

**Additional information**
The configuration for the war name is: (helloworld.ui.html.app.war POM File)
```xml
<plugin>
  <artifactId>maven-war-plugin</artifactId>
  <configuration>
    <warName>uihtml</warName>
  </configuration>
</plugin>
```

--- a/jib-maven-plugin/CHANGELOG.md
+++ b/jib-maven-plugin/CHANGELOG.md
@@ -17,6 +17,7 @@ All notable changes to this project will be documented in this file.
 - Now `<containerizingMode>packaged` works as intended with Spring Boot projects that generate a fat JAR. ([#2170](https://github.com/GoogleContainerTools/jib/issues/2170))
 - Now `<containerizingMode>packaged` correctly identifies the packaged JAR generated at a non-default location when configured with the Maven Jar Plugin's `<classifier>` and `<outputDirectory>`. ([#2170](https://github.com/GoogleContainerTools/jib/issues/2170))
 - `jib:buildTar` with `<container><format>OCI` now builds a correctly formatted OCI archive. ([#2124](https://github.com/GoogleContainerTools/jib/issues/2124))
+- Fixed an issue where configuring the `<warName>` property of the Maven WAR plugin does not work. ([#2206](https://github.com/GoogleContainerTools/jib/issues/2206))
 
 ## 1.8.0
 

--- a/jib-maven-plugin/CHANGELOG.md
+++ b/jib-maven-plugin/CHANGELOG.md
@@ -17,7 +17,7 @@ All notable changes to this project will be documented in this file.
 - Now `<containerizingMode>packaged` works as intended with Spring Boot projects that generate a fat JAR. ([#2170](https://github.com/GoogleContainerTools/jib/issues/2170))
 - Now `<containerizingMode>packaged` correctly identifies the packaged JAR generated at a non-default location when configured with the Maven Jar Plugin's `<classifier>` and `<outputDirectory>`. ([#2170](https://github.com/GoogleContainerTools/jib/issues/2170))
 - `jib:buildTar` with `<container><format>OCI` now builds a correctly formatted OCI archive. ([#2124](https://github.com/GoogleContainerTools/jib/issues/2124))
-- Fixed an issue where configuring the `<warName>` property of the Maven WAR plugin does not work. ([#2206](https://github.com/GoogleContainerTools/jib/issues/2206))
+- Fixed an issue where configuring the `<warName>` property of the Maven WAR plugin fails the build. ([#2206](https://github.com/GoogleContainerTools/jib/issues/2206))
 
 ## 1.8.0
 

--- a/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java
+++ b/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java
@@ -220,8 +220,7 @@ public class MavenProjectProperties implements ProjectProperties {
       throws IOException {
     try {
       if (isWarProject()) {
-        Build build = project.getBuild();
-        Path war = Paths.get(build.getDirectory(), build.getFinalName() + ".war");
+        Path war = getWarArtifact();
         Path explodedWarPath = tempDirectoryProvider.newDirectory();
         ZipUtil.unzip(war, explodedWarPath);
         return JavaContainerBuilderHelper.fromExplodedWar(javaContainerBuilder, explodedWarPath);
@@ -417,6 +416,24 @@ public class MavenProjectProperties implements ProjectProperties {
   @Override
   public boolean isOffline() {
     return session.isOffline();
+  }
+
+  @VisibleForTesting
+  Path getWarArtifact() {
+    Build build = project.getBuild();
+    String warName = build.getFinalName();
+
+    Plugin warPlugin = project.getPlugin("org.apache.maven.plugins:maven-war-plugin");
+    if (warPlugin != null) {
+      for (PluginExecution execution : warPlugin.getExecutions()) {
+        if ("default-war".equals(execution.getId())) {
+          Xpp3Dom configuration = (Xpp3Dom) execution.getConfiguration();
+          warName = getChildValue(configuration, "warName").orElse(warName);
+        }
+      }
+    }
+
+    return Paths.get(build.getDirectory(), warName + ".war");
   }
 
   /**

--- a/jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/MavenProjectPropertiesTest.java
+++ b/jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/MavenProjectPropertiesTest.java
@@ -944,6 +944,61 @@ public class MavenProjectPropertiesTest {
         .info("Spring Boot repackaging (fat JAR) detected; using the original JAR");
   }
 
+  @Test
+  public void testGetWarArtifact() {
+    Mockito.when(mockBuild.getDirectory()).thenReturn(Paths.get("/foo/bar").toString());
+    Mockito.when(mockBuild.getFinalName()).thenReturn("helloworld-1");
+
+    Assert.assertEquals(
+        Paths.get("/foo/bar/helloworld-1.war"), mavenProjectProperties.getWarArtifact());
+  }
+
+  @Test
+  public void testGetWarArtifact_warNameProperty() {
+    Mockito.when(mockBuild.getDirectory()).thenReturn(Paths.get("/foo/bar").toString());
+    Mockito.when(mockBuild.getFinalName()).thenReturn("helloworld-1");
+
+    Mockito.when(mockMavenProject.getPlugin("org.apache.maven.plugins:maven-war-plugin"))
+        .thenReturn(mockPlugin);
+    Mockito.when(mockPlugin.getExecutions()).thenReturn(Arrays.asList(mockPluginExecution));
+    Mockito.when(mockPluginExecution.getId()).thenReturn("default-war");
+    Mockito.when(mockPluginExecution.getConfiguration()).thenReturn(pluginConfiguration);
+    addXpp3DomChild(pluginConfiguration, "warName", "baz");
+
+    Assert.assertEquals(Paths.get("/foo/bar/baz.war"), mavenProjectProperties.getWarArtifact());
+  }
+
+  @Test
+  public void testGetWarArtifact_noWarNameProperty() {
+    Mockito.when(mockBuild.getDirectory()).thenReturn(Paths.get("/foo/bar").toString());
+    Mockito.when(mockBuild.getFinalName()).thenReturn("helloworld-1");
+
+    Mockito.when(mockMavenProject.getPlugin("org.apache.maven.plugins:maven-war-plugin"))
+        .thenReturn(mockPlugin);
+    Mockito.when(mockPlugin.getExecutions()).thenReturn(Arrays.asList(mockPluginExecution));
+    Mockito.when(mockPluginExecution.getId()).thenReturn("default-war");
+    Mockito.lenient().when(mockPluginExecution.getConfiguration()).thenReturn(pluginConfiguration);
+
+    Assert.assertEquals(
+        Paths.get("/foo/bar/helloworld-1.war"), mavenProjectProperties.getWarArtifact());
+  }
+
+  @Test
+  public void testGetWarArtifact_executionIdNotMatched() {
+    Mockito.when(mockBuild.getDirectory()).thenReturn(Paths.get("/foo/bar").toString());
+    Mockito.when(mockBuild.getFinalName()).thenReturn("helloworld-1");
+
+    Mockito.when(mockMavenProject.getPlugin("org.apache.maven.plugins:maven-war-plugin"))
+        .thenReturn(mockPlugin);
+    Mockito.when(mockPlugin.getExecutions()).thenReturn(Arrays.asList(mockPluginExecution));
+    Mockito.when(mockPluginExecution.getId()).thenReturn("no-id-match");
+    Mockito.lenient().when(mockPluginExecution.getConfiguration()).thenReturn(pluginConfiguration);
+    addXpp3DomChild(pluginConfiguration, "warName", "baz");
+
+    Assert.assertEquals(
+        Paths.get("/foo/bar/helloworld-1.war"), mavenProjectProperties.getWarArtifact());
+  }
+
   private BuildContext setUpBuildContext(String appRoot, ContainerizingMode containerizingMode)
       throws InvalidImageReferenceException, IOException, CacheDirectoryCreationException {
     JavaContainerBuilder javaContainerBuilder =

